agreement/voteBatch.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package agreement
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/committee"
+	"github.com/algorand/go-algorand/util/execpool"
+)
+
+type voteBatchProcessor struct {
+	outChan chan<- interface{}
+}
+
+// ProcessBatch implements stream.BatchProcessor interface
+func (vbp *voteBatchProcessor) ProcessBatch(jobs []execpool.InputJob) {
+	verificationTasks := makeVerificationTasksAndResults(len(jobs))
+	// taskedJobs are the jobs that passed the initial checks and a signiture verification task if created for them
+	taskedJobs := make([]*asyncVerifyVoteRequest, 0, len(jobs))
+	for i := range jobs {
+		req := jobs[i].(*asyncVerifyVoteRequest)
+		select {
+		case <-req.ctx.Done():
+			// request of the job is cancelled, return an error response on the channel
+			vbp.GetErredUnprocessed(jobs[i], req.ctx.Err())
+		default:
+		}
+		// if this is an eq vote
+		if req.uev != nil {
+			vts, m, err := req.uev.getEquivocVerificationTasks(req.l)
+			if err != nil {
+				var e *LedgerDroppedRoundError
+				cancelled := errors.As(err, &e)
+				vbp.outChan <- &asyncVerifyVoteResponse{index: req.index, message: req.message, err: err, cancelled: cancelled, req: req}
+				continue
+			}
+			taskedJobs = append(taskedJobs, req)
+			verificationTasks.addEqVoteJob(vts, m, req.uev)
+			continue
```

Comment:
nit: could you avoid the continue statements by using an else case?
I think it may be easier to understand

Reply:
There are 4 `continue` statements in this loop. Replacing them with if/else will create many more branches and nesting. You think it will still be easier? 
Now, the concept is simple: if error or not the type, move to the next one.